src/artifact_providers/base.ts

```diff
@@ -199,7 +199,7 @@ export abstract class BaseArtifactProvider {
    * @param downloadDirectory Directory where downloaded artifact is stored
    * @returns Absolute path to the saved file
    */
-  protected abstract async doDownloadArtifact(
+  protected abstract doDownloadArtifact(
```

Comment:
Could you clarify why `async` was removed?

Reply:
It's an abstract method, so it shouldn't force the methods overriding it to be `async` since there are other ways to to return a promise. Also, to make Typescript happy (error TS1243).